modern/src/utils.js

```diff
@@ -1,5 +1,9 @@
 import { xml2js } from "xml-js";
 
+export function isPromise(obj) {
+  return "function" == typeof obj.then;
```

Comment:
I’d swap the order of these and use `===`.

https://en.m.wikipedia.org/wiki/Yoda_conditions

Reply:
sounds good